### PR TITLE
fix(reporters): `summary` to intercept logger's streams even when they are not `process.std*` streams

### PR DESCRIPTION
**Pull request by AriPerkkio**
### Description

While adding test cases for summary reporter, I noticed bug around window rendered. It's really edge-case bug that happens only when using Node API and passing custom `stdout`/`stderr` while using summary reporter.

So this is more of a `test(reporter): summary reporter tests` than a `fix:`, but fix has more priority.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.
- [x] Please check [Allow edits by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to make review process faster. Note that this option is not available for repositories that are owned by Github organizations.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: packages/vitest/src/node/reporters/renderers/windowedRenderer.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import type { Writable } from 'node:stream'
 import type { Vitest } from '../../core'
 import { stripVTControlCharacters } from 'node:util'
 
-const DEFAULT_RENDER_INTERVAL_MS = 1_000
+const DEFAULT_RENDER_INTERVAL_MS = 100
```
Comment (AriPerkkio) [inline packages/vitest/src/node/reporters/renderers/windowedRenderer.ts]: This value was never used 🙃 

https://github.com/vitest-dev/vitest/blob/fab1b6020bb1118b52308bdf02f4fef0567f347a/packages/vitest/src/node/reporters/renderers/windowedRenderer.ts#L89-L91

File: packages/vitest/src/node/reporters/renderers/windowedRenderer.ts [hunk 2]
```diff
@@ -48,6 +49,8 @@ export class WindowRenderer {
     this.cleanups.push(
       this.interceptStream(process.stdout, 'output'),
       this.interceptStream(process.stderr, 'error'),
+      this.interceptStream(options.logger.outputStream, 'output'),
+      this.interceptStream(options.logger.errorStream, 'error'),
```
Comment (AriPerkkio) [inline packages/vitest/src/node/reporters/renderers/windowedRenderer.ts]: This part fixes a bug where virtual buffer would not capture `logger.log` calls that were made by other reporters when custom `VitestOptions.stdout` was used.

File: packages/vitest/src/node/reporters/renderers/windowedRenderer.ts (hunk 1 above)
Comment (AriPerkkio) [reply packages/vitest/src/node/reporters/renderers/windowedRenderer.ts]: Oh actually it is used here, for a bit different purpose.

https://github.com/vitest-dev/vitest/blob/fab1b6020bb1118b52308bdf02f4fef0567f347a/packages/vitest/src/node/reporters/renderers/windowedRenderer.ts#L63

Need to add another (internal) option.

File: packages/vitest/src/node/reporters/summary.ts [hunk 3]
```diff
@@ -12,6 +12,12 @@ const FINISHED_TEST_CLEANUP_TIME_MS = 1_000
 
 interface Options {
   verbose?: boolean
+
+  /** @internal */
+  interval?: ConstructorParameters<typeof WindowRenderer>[0]['interval']
+
+  /** @internal */
+  threshold?: ConstructorParameters<typeof WindowRenderer>[0]['threshold']
```
Comment (hi-ogawa) [inline packages/vitest/src/node/reporters/summary.ts]: nit. this looks mildly wild. Why not export something like `WindowRendererOptions`?

Comment (AriPerkkio) [reply packages/vitest/src/node/reporters/summary.ts]: I would not expose name `Window` from summary reporter's API. It's supposed to be implementation detail. Any other naming suggestions?

Comment (AriPerkkio) [reply packages/vitest/src/node/reporters/summary.ts]: Oh this is about just exporting the type, not about options structure. Now using `WindowRendererOptions` here.

**Review by hi-ogawa — APPROVED** [review]
